test/reductions/thomasvandoren/test/TestMini.future

```diff
@@ -0,0 +1,5 @@
+bug: user defined reduction accumulate() with tuple arg fails to resolve
+
+When accumulate() accepts a tuple, e.g. accumulate(t: (eltType, int)), the
+compiler fails to resolve it. The compiler does not appear to be realizing the
+first item in the tuple is a type.
```

Comment:
I'd word this differently: The issue seems to be related to a bad interaction between tuples and references.  Specifically, the compiler is passing a ref-to-eltType into the routine as the first tuple element, but due to the way tuples are handled, it doesn't seem to be resolving to the routine that simply wants an eltType as its first tuple element (and we don't have a way to declare that it should be ref-to-eltType, nor am I certain we would want to).